app/src/test/java/com/mapzen/erasermap/view/MainActivityTest.kt

```diff
@@ -248,6 +257,17 @@ public class MainActivityTest {
     }
 
     @Test
+    public fun showRoutePreview_shouldAddStartandEndPins() {
+        activity.showRoutePreview(getTestLocation(), getTestFeature())
+        activity.success(TestRoute())
+        Robolectric.flushForegroundThreadScheduler()
+        val shadowStartPin = ShadowExtractor.extract(activity.startPin) as ShadowMapData
+        val shadowEndPin = ShadowExtractor.extract(activity.startPin) as ShadowMapData
+        assertThat(shadowEndPin.points).isNotNull()
```

Comment:
Let's add an assertion here to verify the start and end pins are actually added to `ShadowTangram.dataSources` as well.

For bonus points we could also verify the lat and long values of the pins match the test location and test feature :)